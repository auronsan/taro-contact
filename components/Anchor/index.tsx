import React, { ReactNode } from 'react';
import clsx from 'clsx';
import classes from './anchor.module.css';

type AnchorProps = {
  href: string;
  children: ReactNode;
  // font-size
  size?: 'inherit' | 'xs' | 'sm' | 'md' | 'lg';

  className?: string;
  style?: React.CSSProperties;

  id?: string;
};

export const Anchor: React.FC<AnchorProps> = (props: AnchorProps) => {
  const { href, children, size, className, style, id } = props;
  return (
    <a
      href={href}
      className={clsx(className, { [classes.root]: true, [classes[`fs-${size}`]]: !!size })}
      style={style}
      id={id}
      data-testid={id}
    >
      {children}
    </a>
  );
};
