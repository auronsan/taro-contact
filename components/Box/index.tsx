import { MouseEventHandler } from 'react';

export type BoxProps = {
  /** Class added to the root element, if applicable */
  className?: string;
  /** Inline style added to root component element*/
  style?: React.CSSProperties;
  component?: keyof JSX.IntrinsicElements;
  children?: React.ReactNode;

  // flexbox
  flex?: number;

  // margin
  m?: number;

  // margin-top
  mt?: number;

  // padding
  p?: number;

  // padding-x
  px?: number;

  // padding-y
  py?: number;

  // onclick
  onClick?: MouseEventHandler;

  // role
  role?: string;

  id?: string;
};

export const Box = ({
  children,
  className = '',
  component = 'div',
  flex,
  style,
  m,
  p,
  px,
  py,
  mt,
  role,
  id,
  ...rest
}: BoxProps) => {
  const Component = component;

  return (
    <Component
      {...(className ? { className } : {})}
      role={role}
      id={id}
      data-testid={id}
      style={{
        flex,
        ...style,
        ...(typeof m === 'number' ? { margin: `${m}px` } : {}),
        ...(typeof p === 'number' ? { padding: `${p}px` } : {}),
        ...(typeof px === 'number' ? { paddingLeft: `${px}px`, paddingRight: `${px}px` } : {}),
        ...(typeof py === 'number' ? { paddingTop: `${py}px`, paddingBottom: `${py}px` } : {}),
        ...(typeof mt === 'number' ? { marginTop: `${mt}px` } : {}),
      }}
      {...rest}
    >
      {children}
    </Component>
  );
};
