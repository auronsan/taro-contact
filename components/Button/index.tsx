import { DOMAttributes } from 'react';
import clsx from 'clsx';
import classes from './button.module.css';

export type ButtonProps = DOMAttributes<HTMLButtonElement> & {
  className?: string;
  children: React.ReactNode;
  // color
  c?: 'primary' | 'red' | 'yellow';

  // variant
  v?: 'outline' | 'filled';

  // flex
  flex?: number;

  style?: React.CSSProperties;

  id?: string;

  type?: 'button' | 'submit';
};

export const Button = ({
  children,
  className,
  c = 'primary',
  v = 'filled',
  flex,
  style,
  id,
  type = 'button',
  ...rest
}: ButtonProps) => (
  <button
    type={type === 'submit' ? 'submit' : 'button'}
    className={clsx(
      classes.root,
      {
        [classes[`button-${c}`]]: !!c,
        [classes[`button-variant-${v}`]]: !!v,
      },
      className
    )}
    style={{
      ...(flex ? { flex } : {}),
      ...style,
    }}
    data-testid={id}
    id={id}
    {...rest}
  >
    {children}
  </button>
);
