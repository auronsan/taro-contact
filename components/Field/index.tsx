import React from 'react';
import { UseFormReturnType } from '@mantine/form';

export type FormInputProps = {
  prefix?: string;
  fieldName: string;
  form: UseFormReturnType<any>;
  children: (props: { value: any; onChange: any }) => React.ReactElement;
};
export const Field = ({
  fieldName,
  form,
  prefix = '',
  children,
}: FormInputProps): React.ReactElement => {
  const prefixFieldName = prefix ? `${prefix}.${fieldName}` : `${fieldName}`;
  const { value = '', onChange } = form.getInputProps(prefixFieldName);
  return children({ value, onChange });
};
