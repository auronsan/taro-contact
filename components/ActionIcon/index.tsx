import { Box } from '@/components/Box';
import classes from './actionIcon.module.css';

type TActionIconProps = {
  onClick?: () => void;
  children?: React.ReactNode;
  id?: string;
};
export const ActionIcon = (props: TActionIconProps): React.ReactElement => {
  const { onClick, children, id } = props;

  return (
    <Box
      className={classes.root}
      onClick={(e) => {
        e.stopPropagation();
        e.preventDefault();
        if (typeof onClick === 'function') {
          onClick();
        }
      }}
      id={id}
      role="button"
    >
      {children}
    </Box>
  );
};
