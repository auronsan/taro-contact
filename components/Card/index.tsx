import React from 'react';
import { Box } from '@/components/Box';
import classes from './card.module.css';

/**
 * Props for the Card component.
 */
type TCardProps = {
  /** The content to be rendered inside the card. */
  children: React.ReactNode;
};

/**
 * Card component.
 * @param props - The props for the component.
 * @returns The rendered card.
 */
export const Card = ({ children }: TCardProps): React.ReactElement => (
  <Box className={classes.card}>
    <Box>{children}</Box>
  </Box>
);
