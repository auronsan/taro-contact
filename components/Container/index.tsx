import clsx from 'clsx';
import type { BoxProps } from '@/components/Box';
import { Box } from '@/components/Box';
import classes from './container.module.css';

export interface ContainerProps extends BoxProps {
  /** Sets `max-width` of the container, value is not responsive â€“ it is the same for all screen sizes. Numbers are converted to rem. Ignored when `fluid` prop is set. `'md'` by default */
  size?: (string & {}) | number;

  /** Determines whether the container should take 100% of its parent width. If set, `size` prop is ignored. `false` by default. */
  fluid?: boolean;
}

export const Container = (props: ContainerProps) => {
  const { fluid, ...rest } = props;
  return (
    <Box
      {...rest}
      className={clsx({
        [classes.root]: true,
        [classes.fluid]: !!fluid,
      })}
    />
  );
};
