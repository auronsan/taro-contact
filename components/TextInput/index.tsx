import { DetailedHTMLProps, forwardRef, InputHTMLAttributes, Ref } from 'react';
import { Box, BoxProps } from '@/components/Box';
import classes from './textinput.module.css';

export type TextInputProps = DetailedHTMLProps<
  InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
> &
  BoxProps & {
    label?: string;
    id?: string;
    name?: string;
  };
export const TextInput = forwardRef((props: TextInputProps, _ref: Ref<HTMLInputElement>) => {
  const { label, id, name, ...rest } = props;

  return (
    <Box className={classes.root}>
      {label && (
        <label className={classes.label} htmlFor={id}>
          {label}
        </label>
      )}
      <input className={classes.input} name={name} ref={_ref} data-testid={id} id={id} {...rest} />
    </Box>
  );
});
