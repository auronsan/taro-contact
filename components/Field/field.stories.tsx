import React from 'react';
import { useForm } from '@mantine/form';
import { TextInput } from '@/components/TextInput';
import { Field } from './index';

export default {
  title: 'components/Field',
  component: Field,
};

export const withTextInput = () => {
  const form = useForm({
    initialValues: {
      test: '',
    },
  });

  return (
    <Field form={form} fieldName="test">
      {({ value, onChange }) => (
        <TextInput
          data-testid="field"
          value={`${value || ''}`}
          onChange={(e) => {
            const val = e.target.value;
            if (typeof onChange === 'function') {
              onChange(val);
            }
          }}
        />
      )}
    </Field>
  );
};
