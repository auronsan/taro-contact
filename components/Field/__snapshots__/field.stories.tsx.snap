// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Field withTextInput smoke-test 1`] = `
<main class="roboto-normal">
  <div class="textinput_root__IJOrE">
    <input class="textinput_input__ICHQz"
           data-testid="field"
           value
    >
  </div>
  <div class="stack_root__KVCbK toast_toast-container__8fZtQ stack_gap-xs__xTnny"
       id="toast-container"
       data-testid="toast-container"
       style="padding: 0px;"
  >
  </div>
</main>
`;
